### ISSUE TITLE
$similarCounts not sanity checked on line 162 of src/services/Similar.php

### ISSUE THREAD

**Issue by jamiewhitemccann** [nystudio107/craft-similar#29]
### Describe the bug
2021-03-09 10:58:03 [-][3][de99c010b4cfe6c82fedeef4c2b2ddb0][error][yii\base\ErrorException:8] yii\base\ErrorException: Undefined index: 1-175221 in /app/vendor/nystudio107/craft-similar/src/services/Similar.php:162

### To reproduce
We are seeing fatal errors when using this plugin, although we are not 100% on the exact scenario to recreate.

### Expected behaviour
The plugin should not throw error exceptions in this manner

### Versions
- Plugin version: 1.1.0
- Craft version: 3.5.19.1

Suggested patch to sanity check the $similarCounts array using a simple isset on the index:

```
From 3767b2c3d1393956a9d3fbf199e9fcfd5978495b Mon Sep 17 00:00:00 2001
From: Jamie White
Date: Tue, 9 Mar 2021 16:27:27 +0000
Subject: [PATCH] Adding a sanity check for for the $similarCounts array

Fix for the errors we are seeing:
2021-03-09 16:01:29 [-][-][0e5667ba12611515c899ffcdf5691c09][error][yii\base\ErrorException:8] yii\base\ErrorException: Undefined index: 1-175221 in /srv/live/src/hiq/tyre-directory/vendor/nystudio107/craft-similar/src/services/Similar.php:162
---
 src/services/Similar.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/services/Similar.php b/src/services/Similar.php
index 1c5a629..ddd9780 100644
--- a/src/services/Similar.php
+++ b/src/services/Similar.php
@@ -159,7 +159,9 @@ class Similar extends Component
         foreach ($elements as $element) {
             // The `count` property is added dynamically by our CountBehavior behavior
             /** @noinspection PhpUndefinedFieldInspection */
-            $element->count = $similarCounts[$element->siteId . '-' . $element->id];
+            if (isset($similarCounts[$element->siteId . '-' . $element->id])) {
+                $element->count = $similarCounts[$element->siteId . '-' . $element->id];
+            }
         }
 
         return $elements;
-- 
2.27.0

```

**Comment by khalwat** [comment]
Fixed in https://github.com/nystudio107/craft-similar/commit/0d81f2f27cb698c469aefce68ad76379153ced0d

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-similar": "dev-develop as 1.1.1”,
```
Then do a `composer update`

**Comment by jamiewhitemccann** [comment]
Thank you very much Andrew!

Regards,
Jamie


[signature_1389440161]

Jamie White
Technical Lead
McCann, Tower Wharf, Cheese Lane, Bristol, BS2 0JJ
0117 921 8102
www.mccannbristol.co.uk<http://www.mccannbristol.co.uk/>

**Comment by solutiondrop** [comment]
Hi,

I was receiving the same error and set my semver and updated with fix. It works as expected when I've assigned tags to a product, but when there are no tags assigned it now show products that aren't tagged similarly. It used to show nothing, but now with the update it shows the most recently added products.

Here's how I'm calling them:

`{% set limitCriteria = craft.products.limit(20) %}`
` {% set similarEntriesByTags = craft.similar.find({ element: product, context: product.recommendationTags, criteria: limitCriteria }) %}` 
` ...` 
` {% for similarEntry in similarEntriesByTags %}` 
` ...` 
` {% endfor %}`

I'm wondering if I'm doing something wrong here? Thanks!

**Comment by jamiewhitemccann** [comment]
My college has opened this ticket, which I think is actually the root cause of the error firing in the first place:

https://github.com/nystudio107/craft-similar/issues/31

**Comment by khalwat** [comment]
This issue is fixed in 1.1.1, which is released. https://github.com/nystudio107/craft-similar/releases/tag/1.1.1

@solutiondrop If you have a separate issue, please file a new issue. Might be related to: https://github.com/nystudio107/craft-similar/issues/31
